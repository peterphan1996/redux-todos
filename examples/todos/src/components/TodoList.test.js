import React from 'react';
import { shallow } from 'enzyme';

import TodoList from './TodoList';

const setup = setupProps => {
  const defaultProps = {
    todos: [],
    onTodoClick: jest.fn(),
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(<TodoList todos={props.todos} toggleTodo={props.onTodoClick} />);

  return {
    props,
    wrapper,
  };
};

describe('TodoList', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  describe('with todos', () => {
    const { wrapper, props } = setup({
      todos: [
        {
          text: 'Codelink',
          completed: false,
          id: 0,
        },
        {
          text: 'Nhat',
          completed: true,
          id: 0,
        },
      ],
    });

    test('renders a list of todos', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('calls onTodoClick when its todo children clicked', () => {
      wrapper
        .find('Todo')
        .at(0)
        .simulate('click');
      expect(props.onTodoClick).toHaveBeenCalledTimes(1);
    });
  });
})