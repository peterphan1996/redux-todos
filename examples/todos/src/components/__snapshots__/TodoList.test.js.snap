// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders without crashing 1`] = `<ul />`;

exports[`TodoList with todos renders a list of todos 1`] = `
<ul>
  <Todo
    completed={false}
    id={0}
    key="0"
    onClick={[Function]}
    text="Codelink"
  />
  <Todo
    completed={true}
    id={0}
    key="0"
    onClick={[Function]}
    text="Nhat"
  />
</ul>
`;
